src/main/java/at/codersbay/datastructures/Fractional.java

```diff
@@ -1,74 +1,146 @@
 package at.codersbay.datastructures;
 
+import java.util.Scanner;
+
 /**
  * A Fractional is a class that consists of a numerator and a denominator
  * e.g.: 1/2, 1/3, 24/37, ..
  */
 public class Fractional {
 
+    public Integer numerator;
+    public Integer denominator;
+
     public Fractional(Integer numerator, Integer denominator) {
+
+            this.numerator = numerator;
+            this.denominator = denominator;
+
+            if (denominator == 0){
+            throw new IllegalArgumentException("You shall not divide through zero!");
+        } else if (denominator == null){
+            throw new IllegalArgumentException();
+        }
+
         // TODO implement to fix tests in FractionalTest1
     }
 
+
     public Integer getNumerator() {
+        return this.numerator;
         // TODO implement to fix tests in FractionalTest1
-        return -1;
     }
 
     public Integer getDenominator() {
+        return this.denominator;
         // TODO implement to fix tests in FractionalTest1
-        return -1;
     }
 
     public Float asFloat() {
+        Float asFloatIntOne = this.numerator.floatValue();
+        Float asFloatIntTwo = this.denominator.floatValue();
+                ;
+        Float asFloatFloat = asFloatIntOne/asFloatIntTwo;
         // TODO implement to fix tests in FractionalTest2
-        return -1f;
+        return asFloatFloat;
     }
 
     @Override
     public boolean equals(Object obj) {
+
+            if(obj instanceof Fractional) {
+                Fractional frac = (Fractional) obj;
+
+                if(((double) this.numerator / (double) this.denominator) == ((double) frac.numerator / (double) frac.denominator)) {
+                    return true;
+                }
+                return denominator * frac.numerator == numerator * frac.denominator;
+            }
         // TODO implement to fix tests in FractionalTest3
-        return false;
+            return false;
     }
 
     public Fractional multiply(Fractional other) {
+        if (((this.numerator < 0 && other.numerator < 0) && (this.denominator < 0 && other.denominator < 0)) || ((this.numerator > 0 && other.numerator > 0) && (this.denominator > 0 && other.denominator > 0))) {
+            return new Fractional(this.numerator * other.numerator, this.denominator * other.denominator);
+        } else if (this.numerator < 0){
+            this.numerator *= -1;
```

Comment:
hä ? ❓ was ist wenn numrator aber -5 ist - dann will ich ja nicht mit -1 multiplizieren. 

generell ist es dem multiply eigentlich egal welche kombi von positiven/negativen zahlen du hast - du machst eigentlich immer einfach nur `return new Fractional(this.numerator * other.numerator, this.denominator * other.denominator);`